python/ray/tune/logger/json.py

```diff
@@ -60,7 +60,7 @@ def close(self):
         self.local_out.close()
 
     def update_config(self, config: Dict):
-        self.config = config
+        self.config = config if isinstance(config, dict) else config.to_dict()
```

Comment:
Nit: Can we actually make this more save and instead check for: `... to_dict() if isinstance(..., AlgorithmConfig) else config`?
Otherwise, if the user would provide anything unknown, we would get a strange error here.